src/Microsoft.DotNet.Wpf/src/PresentationFramework/Microsoft/Win32/OpenFolderDialog.cs

```diff
@@ -0,0 +1,326 @@
+        {
+            base.PrepareDialog(dialog);
+
+            dialog.SetFileName(CriticalFileName);
+        }
+
+        private protected override bool HandleFileOk(IFileDialog dialog)
+        {
+            if (!base.HandleFileOk(dialog))
+            {
+                return false;
+            }
+
+            string[] saveFileNames = _fileNames;
+            bool ok = false;
+
+            try
+            {
+                _fileNames = ProcessFolders(dialog);
+
+                var cancelArgs = new CancelEventArgs();
+                OnFileOk(cancelArgs);
+                ok = !cancelArgs.Cancel;
+            }
+            finally
+            {
+                if (!ok)
+                {
+                    _fileNames = saveFileNames;
+                }
+            }
+            return ok;
+        }
+
+        #endregion Internal Methods
+
+        //---------------------------------------------------
+        //
+        // Internal Properties
+        //
+        //---------------------------------------------------
+        //#region Internal Properties
+        //#endregion Internal Properties
+
+        //---------------------------------------------------
+        //
+        // Internal Events
+        //
+        //---------------------------------------------------
+        //#region Internal Events
+        //#endregion Internal Events
+
+        //---------------------------------------------------
+        //
+        // Private Methods
+        //
+        //---------------------------------------------------
+        #region Private Methods
+
+        //  Provides the actual implementation of initialization tasks.
+        //  Initialize() is called from both the constructor and the
+        //  public Reset() function to set default values for member
+        //  variables and for the options bitmask.
+        //
+        //  We only perform OpenFolderDialog() specific reset tasks here;
+        //  it's the calling code's responsibility to ensure that the
+        //  base is initialized first.
+        //
+        private void Initialize()
+        {
+            // FOS_FILEMUSTEXIST
+            // Folder must exist. Otherwise, the folder name remaining in
+            // text box might be returned as a nested folder. This flag is
+            // now enforced by FOS_PICKFOLDERS.
+            SetOption(FOS.FILEMUSTEXIST, true);
+
+            // FOS_PICKFOLDERS
+            // This is a folder selecting dialog.
+            SetOption(FOS.PICKFOLDERS, true);
+        }
+
+        #endregion Private Methods
+
+        //---------------------------------------------------
+        //
+        // Private Properties
+        //
+        //---------------------------------------------------
+        #region Private Properties
+
+        /// <summary>
+        ///  Gets a string containing the full path of the file selected in 
+        ///  the file dialog box.
```

Comment:
Should replace the `file` to `folder`?

```suggestion
        ///  the folder dialog box.
```